### ISSUE TITLE
vmselect: `search.skipSlowReplicas` isn't respected when globalReplicationFactor > 1

### ISSUE THREAD

**Issue by hagen1778** [VictoriaMetrics/VictoriaMetrics#6924]
### Describe the bug

Let's define the following vmselect configuration:
```
./bin/vmselect 
  -storageNode=gr1/:8211,gr1/:8212 
  -storageNode=gr2/:8213,gr2/:8214 
  -search.skipSlowReplicas=true
  -globalReplicationFactor=2
```
Here we have two vmstorage groups and `-globalReplicationFactor=2`, which effectively means that "every ingested sample is replicated across multiple vmstorage groups". Hence, `gr1` and `gr2` contain identical data set. And when we set `-search.skipSlowReplicas=true` it is expected vmselect should return result as soon as at least one storage group returned the full result.
In current state, `-search.skipSlowReplicas` is ignored on the storage group level. It is only respected within the group (with `-replicationFactor` flag).

### To Reproduce

Start vmselect with mentioned config, but add artificial delay to responses for nodes in one storage group. Run range query with trace enabled and observe the output. The trace should contain subtraces showing that vmselect have waited for all 4 storages to respond. 
If `globalReplicationFactor` would have been respected, vmselect should have cancel requests to slow nodes in one of the storage groups.

Please see this code https://github.com/VictoriaMetrics/VictoriaMetrics/blob/1aa9f7be4e9d058857752d1531ff9d22d0090268/app/vmselect/netstorage/netstorage.go#L1907-L1925
vmselect checks for amount of responses within the group, but doesnt check for number of groups to respond. Probably, the naive fix will look like the following:
```
		resultsCollectedPerGroup[group]++
		if *skipSlowReplicas && (len(resultsCollectedPerGroup) == groupsCount || len(resultsCollectedPerGroup) > groupsCount-*globalReplicationFactor) {
			canSkipSlowReplicas := true
```

### Version

all versions with globalReplicationFactor support

### Used command-line flags

```
./bin/vmselect 
  -storageNode=gr1/:8211,gr1/:8212 
  -storageNode=gr2/:8213,gr2/:8214 
  -search.skipSlowReplicas=true
  -globalReplicationFactor=2
```

**Comment by hagen1778** [comment]
Depends on https://github.com/VictoriaMetrics/VictoriaMetrics/issues/7199

**Comment by f41gh7** [comment]
The bugfix was included into [v1.108.0](https://docs.victoriametrics.com/changelog/#v11080) and [v1.102.9](https://docs.victoriametrics.com/changelog/#v11029)
